galley/pkg/config/analysis/analyzers/auth/servicerolebindings.go

```diff
@@ -0,0 +1,38 @@
+package auth
+
+import (
+	"istio.io/api/rbac/v1alpha1"
+
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/galley/pkg/config/processor/metadata"
+	"istio.io/istio/galley/pkg/config/resource"
+)
+
+// ServiceRoleBindingAnalyzer checks the validity of service role bindings
+//TODO: Unit tests
+type ServiceRoleBindingAnalyzer struct{}
+
+var _ analysis.Analyzer = &ServiceRoleBindingAnalyzer{}
+
+// Name implements Analyzer
+func (s *ServiceRoleBindingAnalyzer) Name() string {
+	return "auth.ServiceRoleBindingAnalyzer"
+}
+
+// Analyze implements Analyzer
+func (s *ServiceRoleBindingAnalyzer) Analyze(ctx analysis.Context) {
+	ctx.ForEach(metadata.IstioRbacV1Alpha1Servicerolebindings, func(r *resource.Entry) bool {
+		s.analyzeRoleBinding(r, ctx)
+		return true
+	})
+}
+
+func (s *ServiceRoleBindingAnalyzer) analyzeRoleBinding(r *resource.Entry, ctx analysis.Context) {
+	srb := r.Item.(*v1alpha1.ServiceRoleBinding)
+	ns, _ := r.Metadata.Name.InterpretAsNamespaceAndName()
+
+	if ctx.Find(metadata.IstioRbacV1Alpha1Serviceroles, resource.NewName(ns, srb.RoleRef.Name)) == nil {
```

Comment:
Suggestion: We can add a new method to context (i.e. Exists) that returns bool. This way, this code gets slightly easier to read.

Also, I'd suggest hoisting the service role name to a local variable to make it look a bit cleaner.
```suggestion
        role := resource.NewName(ns, binding.RoleRef.Name)
	if !ctx.Exists(metadata.IstioRbacV1Alpha1Serviceroles, role) {
```